drivers/storage/port/storport/miniport.c

```diff
@@ -82,8 +82,10 @@ InitializeConfiguration(
                       PortConfig->NumberOfAccessRanges * sizeof(ACCESS_RANGE));
     }
 
-    for (i = 0; i < 7; i++)
-        PortConfig->InitiatorBusId[i] = 0xff;
+    for (i = 0; i < RTL_NUMBER_OF(PortConfig->InitiatorBusId); i++)
```

Comment:
Hmm.. `RTL_NUMBER_OF(PortConfig->InitiatorBusId)` is equal to `RTL_NUMBER_OF(ConfigInfo->InitiatorBusId)` in the freeldr? Here it was a `< 7` check, and there it was `< 8`.

Reply:
Precisely. The InitiatorBusId array of the SCSI/STORPORT `PORT_CONFIGURATION_DATA` structure has a certain number of elements (8). Why would we then initialize only the seven first ones and not the eight, in storport? (while all 8 are initialized in scsiport and freeldr).
Look also at that suspiciously similarly-sounding old MS KB report: https://www.betaarchive.com/wiki/index.php/Microsoft_KB_Archive/943545